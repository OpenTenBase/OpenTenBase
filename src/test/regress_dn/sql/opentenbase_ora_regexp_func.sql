
\c regression_ora

-- test regexp_count(numeric, numeric)
select regexp_count(123423423466722341234, 123) from dual;
select regexp_count('123423423466722341234', '123') from dual;

select regexp_count(123423423466722341234, 123, 3) from dual;
select regexp_count('123423423466722341234', '123', 3) from dual;

select regexp_count(123423423466722341234, 123, 3, 'x') from dual;
select regexp_count('123423423466722341234', '123', 3, 'x') from dual;

-- test regexp_count(text, numeric)
select regexp_count('12341234', 34) from dual;
select regexp_count('12341234', 34, 3) from dual;
select regexp_count('12341234', 34, 4) from dual;
select regexp_count('12341234', 34, 1, 'x') from dual;

-- test regexp_count(numeric, text)
select regexp_count(12341234, '34') from dual;
select regexp_count(12341234, '34', 3) from dual;
select regexp_count(12341234, '34', 4) from dual;
select regexp_count(12341234, '34', 1, 'x') from dual;

create table regexp_tbl (a number, b text, c numeric, d text);
insert into regexp_tbl values(1234512378123, '1234512378123', 123, '123');
insert into regexp_tbl values(1237812378123, '1237812378123', 78, '78');
insert into regexp_tbl values(12256223234123, '12256223234123', 23, '23');

select regexp_count(a, c) from regexp_tbl;
select regexp_count(a, c, 3) from regexp_tbl;
select regexp_count(a, c, 3, 'x') from regexp_tbl;

select regexp_count(a, d) from regexp_tbl;
select regexp_count(a, d, 3) from regexp_tbl;
select regexp_count(a, d, 3, 'x') from regexp_tbl;

select regexp_count(b, c) from regexp_tbl;
select regexp_count(b, c, 3) from regexp_tbl;
select regexp_count(b, c, 3, 'x') from regexp_tbl;

drop table regexp_tbl;

-- BEGIN: tapd 119561653 
drop table if exists regexp_count_tb1_toast ;
create table regexp_count_tb1_toast(id int,id1 text);
insert into regexp_count_tb1_toast values(3,'adbadrXXbcdfgdiec,ncdn');
insert into regexp_count_tb1_toast values(1,'adbadrabmmcdfBBBbb,ncdn'||repeat('bx-',300000));
insert into regexp_count_tb1_toast values(2,'cdx');
select id,regexp_count(id1,'b',1) from regexp_count_tb1_toast order by 1;
drop table if exists regexp_count_tb1_toast ;
-- END: 119561653
