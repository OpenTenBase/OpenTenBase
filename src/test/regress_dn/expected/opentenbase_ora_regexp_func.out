\c regression_ora
-- test regexp_count(numeric, numeric)
select regexp_count(123423423466722341234, 123) from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

select regexp_count('123423423466722341234', '123') from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

select regexp_count(123423423466722341234, 123, 3) from dual;
 REGEXP_COUNT 
--------------
            1
(1 row)

select regexp_count('123423423466722341234', '123', 3) from dual;
 REGEXP_COUNT 
--------------
            1
(1 row)

select regexp_count(123423423466722341234, 123, 3, 'x') from dual;
 REGEXP_COUNT 
--------------
            1
(1 row)

select regexp_count('123423423466722341234', '123', 3, 'x') from dual;
 REGEXP_COUNT 
--------------
            1
(1 row)

-- test regexp_count(text, numeric)
select regexp_count('12341234', 34) from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

select regexp_count('12341234', 34, 3) from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

select regexp_count('12341234', 34, 4) from dual;
 REGEXP_COUNT 
--------------
            1
(1 row)

select regexp_count('12341234', 34, 1, 'x') from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

-- test regexp_count(numeric, text)
select regexp_count(12341234, '34') from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

select regexp_count(12341234, '34', 3) from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

select regexp_count(12341234, '34', 4) from dual;
 REGEXP_COUNT 
--------------
            1
(1 row)

select regexp_count(12341234, '34', 1, 'x') from dual;
 REGEXP_COUNT 
--------------
            2
(1 row)

create table regexp_tbl (a number, b text, c numeric, d text);
insert into regexp_tbl values(1234512378123, '1234512378123', 123, '123');
insert into regexp_tbl values(1237812378123, '1237812378123', 78, '78');
insert into regexp_tbl values(12256223234123, '12256223234123', 23, '23');
select regexp_count(a, c) from regexp_tbl;
 REGEXP_COUNT 
--------------
            3
            2
            3
(3 rows)

select regexp_count(a, c, 3) from regexp_tbl;
 REGEXP_COUNT 
--------------
            2
            2
            3
(3 rows)

select regexp_count(a, c, 3, 'x') from regexp_tbl;
 REGEXP_COUNT 
--------------
            2
            2
            3
(3 rows)

select regexp_count(a, d) from regexp_tbl;
 REGEXP_COUNT 
--------------
            3
            2
            3
(3 rows)

select regexp_count(a, d, 3) from regexp_tbl;
 REGEXP_COUNT 
--------------
            2
            2
            3
(3 rows)

select regexp_count(a, d, 3, 'x') from regexp_tbl;
 REGEXP_COUNT 
--------------
            2
            2
            3
(3 rows)

select regexp_count(b, c) from regexp_tbl;
 REGEXP_COUNT 
--------------
            3
            2
            3
(3 rows)

select regexp_count(b, c, 3) from regexp_tbl;
 REGEXP_COUNT 
--------------
            2
            2
            3
(3 rows)

select regexp_count(b, c, 3, 'x') from regexp_tbl;
 REGEXP_COUNT 
--------------
            2
            2
            3
(3 rows)

drop table regexp_tbl;
-- BEGIN: tapd 119561653 
drop table if exists regexp_count_tb1_toast ;
NOTICE:  table "REGEXP_COUNT_TB1_TOAST" does not exist, skipping
create table regexp_count_tb1_toast(id int,id1 text);
insert into regexp_count_tb1_toast values(3,'adbadrXXbcdfgdiec,ncdn');
insert into regexp_count_tb1_toast values(1,'adbadrabmmcdfBBBbb,ncdn'||repeat('bx-',300000));
insert into regexp_count_tb1_toast values(2,'cdx');
select id,regexp_count(id1,'b',1) from regexp_count_tb1_toast order by 1;
 ID | REGEXP_COUNT 
----+--------------
  1 |       300004
  2 |            0
  3 |            2
(3 rows)

drop table if exists regexp_count_tb1_toast ;
-- END: 119561653
